File: backend/src/main/java/com/nishtha/RoomEase/room/service/RoomService.java
package com.nishtha.RoomEase.room.service;

import com.nishtha.RoomEase.room.dto.AddRoomRequest;
import com.nishtha.RoomEase.room.dto.RoomResponse;
import com.nishtha.RoomEase.room.dto.UpdateRoomRequest;

import java.util.List;

public interface RoomService {

    RoomResponse createRoom(Long propertyId,AddRoomRequest request);

    List<RoomResponse> getRoomsByProperty(Long propertyId);

    RoomResponse getRoom(Long roomId);

    RoomResponse updateRoom(Long roomId, UpdateRoomRequest request);

    void deleteRoom(Long roomId);

}

File: frontend/src/services/roomService.js
import api from "./api";

const createRoom = (propertyId, data) =>
    api.post(`/properties/${propertyId}/rooms`, data);

const getRoom=(roomId) =>
    api.get(`/rooms/${roomId}`)

const updateRoom = (roomId, data) =>
    api.put(`/rooms/${roomId}`, data);

const deleteRoom = (roomId) =>
    api.delete(`/rooms/${roomId}`);

const getRoomsByProperty= async (propertyId)=>{
     const reponse= await api.get(`properties/${propertyId}/rooms`);
     return reponse.data;
}

export default {
    createRoom,
    getRoomsByProperty,
    getRoom,
    updateRoom,
    deleteRoom
};